### ISSUE TITLE
Installation fails on raspberry pi

### ISSUE THREAD

**Issue by hferee** [LowMem/kresus_ynh#7]
The install script on a rasperry-pi 3 gives a very high load average (about 25), although not so much CPU usage, and the machine eventually freezes.

I know that this is a low-spec device, and maybe there is nothing you can do to lighten the installation script (this is probably npm's fault), but I definitely think this is the kind of devices yunohost is made for so I'd very much like to see kresus working on it.

FYI, the log only shows this warning before freezing:
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"} (current: {"node":"0.10.29","npm":"1.4.21"})
Node's version is apparently not fresh enough, but if this is really an issue the install script should stop properly with this as an error.

**Comment by LowMem** [comment]
Hello,

Thanks for having reported this issue.

I am surprised that installation freezes on a raspberry pi because I am running it on an [Olinuxino A20 Lime2 device](https://www.olimex.com/Products/OLinuXino/A20/A20-OLinuXIno-LIME2/open-source-hardware) as indicated on [Internet Cube](https://internetcu.be/)/[La Brique Internet](https://labriqueinter.net/) websites…

As you stated, I don't know what to do at software level. :-/
I let this issue open, so that raspberry pi users know about it and/or report their experience and/or make any suggestion of improvement.

**Comment by Phyks** [comment]
Depending on #8, it might be worth trying with a more up to date version of `Node`.

One of the most expensive part of the `npm install` phase is the installation of `pouchdb` which does not offer prebuilt images for ARM (I think), so that it has to be compiled on the Pi, which is quite slow and CPU consuming. This will hopefully not be a blocking issue in the next release as we will be dropping `pouchdb` and Cozy support to move to a more standard SQL database.

**Comment by LowMem** [comment]
Is it better now with the complete rewrite from @Jibec ?

I close this (several months issue), feel free to re-open if problem still occurs or to open a new issue if another problem is on the way to raspberry pi use.
